### PR TITLE
Manage window resizing with Gnome Shell dock (Issue #1350)

### PR DESCRIPTION
**Pull request by frz-dev**
Issue #1350 is due to the fact that Gnome shell dock, currently used by Ubuntu, is not managed.
This causes the window to be badly sized and/or positioned.

This pull request correctly resize/position the Guake window in the presence of a Gnome shell dock.

### CODE REVIEW

File: guake/guake_app.py
Comment (ulidtko) [inline guake/guake_app.py:1013]: `"ubuntu:gnome".lower()` is the same as just `"ubuntu:gnome"`; likewise for `"gnome"`

File: guake/guake_app.py
Comment (ulidtko) [inline guake/guake_app.py:1186]: This is what, **nineteen** nesting levels?..

That's too bad; please see if you can refactor the code into something neater.

File: guake/guake_app.py
Comment (ulidtko) [reply guake/guake_app.py:1013]: I understand that you copied the version comparison logic from Gaetan's Unity-related code. Nevertheless, **casting version strings to float** isn't going to fly very far; that's simply incorrect. With versions, it isn't true that `11.2 > 11.10`. Either add a big fat `FIXME` right here (and there with Unity), or take e.g. [semver lib](https://pypi.org/project/semver/) and do it right.

Comment (ulidtko) [reply guake/guake_app.py:1013]: Next piece of copy-paste. Could you make a variable here?
```python
def is_using_gnome(self):
    linux_distrib = platform.linux_distribution() [1]
    xdg_desktop = os.environ.get('XDG_CURRENT_DESKTOP', '').lower()
    return (distro_before_11_10 and xdg_desktop == "gnome"
          or distro_after_11_10 and xdg_desktop == "ubuntu:gnome")
```

Comment (frz-dev) [reply guake/guake_app.py]: As for version comparison, casting to float is not a problem for Ubuntu as its version number always have a trailing zero in case of one-digit number (so it would be 11.02, not 11.2.)

Anyway, semver has problems with trailing zeroes (it raises a ValueError exception).
I found this [packaging.version](https://packaging.pypa.io/en/latest/version/#packaging.version.parse) module that works just fine (11.04 < 11.10 yields True, 11.2 > 11.10 yields False)

I assume `packaging` should be added to the requirements, but I'm not sure how this is handled.
Should I add this line:  
```packaging = "*"```
to `Pipfile` ?

Comment (frz-dev) [reply guake/guake_app.py]: As for the variable, I think I made a mistake in using `XDG_CURRENT_DESKTOP` in both cases.
It should be `DESKTOP_SESSION` if version is lower than 11.10.

Also, I think that `linux_distrib[1]) - 0.01` was actually meant to actually include 11.10, so I'm changing `<` for `<=`.

Comment (frz-dev) [reply guake/guake_app.py]: I pushed changes to `is_using_gnome`. I'll wait for your feedback to resolve the conversation

File: guake/guake_app.py
Comment (frz-dev) [reply guake/guake_app.py:1190]: Ok, I'll check it tomorrow

Comment (frz-dev) [reply guake/guake_app.py]: I commited the new version with less nesting levels. 
Basically I serialized all the if statements. Don't really like it conceptually, but I guess it's a matter of taste.

Please note that the new code is UNTESTED
(I'm confined at home for the coronavirus, so I'm working on different machine with no double monitor).

File: guake/guake_app.py
Comment (Davidy22) [inline guake/guake_app.py:1010]: Is this function's logic correct? I'm sure there's distros other than ubuntu out there that use GNOME, and I think ubuntu switched back to gnome shell at some point later. Is there no better way to check for gnome shell?

**Review by Davidy22 — CHANGES_REQUESTED** [review]
I don't know enough about gnome shell to comment on the second half with implementation specifics and I can't test this either because I don't use it. Going to need someone else to test and confirm that this code works before merging.

The CI pipeline has been fixed so you'll need to update from master to have our checks run on your branch, and you'll need to get your code to pass our automated tests as well.

Also, would be nice if you could fill in a release notes file describing your fixes, type:

```
make reno SLUG=fix_gnome_shell_window_resize
```

in your shell to open a template in your text editor, delete the sections that aren't relevant to your changes.

**Comment by gsemet** [comment]
can we merge this ticket?

**Comment by Davidy22** [comment]
There's some strange checking logic for detecting gnome shell that I wanted clarification on. I haven't tested this but [this guy did](https://github.com/Guake/guake/pull/1765#issuecomment-920054239) and says this doesn't fix the issue. It's also slightly worrying that the submitter's last messages are that this code is untested, and it stops there.

**Comment by Davidy22** [comment]
We've had a confirmation that this pull request doesn't work, there's still some quality issues and no replies for a while. Closing pr, if you have an updated version of this pull request that does work in the future we can take another look at this.

**Pull request closed without merging**
